src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsClassifier.cs

```diff
@@ -13,29 +14,44 @@ namespace Xamarin.Android.Tasks
 #if ENABLE_MARSHAL_METHODS
 	public sealed class MarshalMethodEntry
```

Comment:
Is there actually need for many of these types to be `public`?  We're not hitting them from a separate unit test assembly, so should they be public at all?

Reply:
The next PR in the series will do all kinds of cleanups like this.